### ISSUE TITLE
Type unification error in parser selection expression

### ISSUE THREAD

**Issue by hanw** [p4lang/p4c#1436]
The following code triggers an error in TypeInference pass.
```
FrontEnd invoking CheckNamedArgs
FrontEnd invoking TypeInference
select-signed.p4(16): error: SelectExpression: Cannot unify bit<16> to int<16>
  return select (latest.khans) {
                 ^^^^^^^^^^^^
```
[select-signed.p4-1.txt](https://github.com/p4lang/p4c/files/2279613/select-signed.p4-1.txt)

**Comment by mbudiu-vmw** [comment]
It looks like the p4 14 converter infers a wrong type for the select case labels.
@ChrisDodd wrote that code.

**Comment by mbudiu-vmw** [comment]
Fixed by #1439
